app/views/errors/show.html.erb

```diff
@@ -22,3 +22,22 @@
     </button>
   </form>
 <% end %>
+
+  <% if @rescue_response == :internal_server_error %>
```

Comment:
@pmcostello these two are awfully similar. Is it possible to create a generic partial for rendering the appropriate error type? We can pair on it to add some tests to this as well if you want?

## Duplicate
This snippet for showing the search functionality for `:not_found` is doing exactly the same thing we need for `:internal_server_error`. We can bring all these together. I have added some suggestions for improvements here (they are not tested).

```
<% if @rescue_response == :not_found %>
  <p class="error-page__search-text"><%= t('.not_found.message') %></p>

  <form data-dough-component="ClearInput" action="/en/search" method="get" role="search" class="error-page__search search">
    <label class="visually-hidden" for="search"><%= t('search_box.label') %></label>
    <input data-dough-clear-input class="search__input" id="search" required type="search" name="query" autocomplete="off" placeholder="<%= t('.not_found.search_placeholder') %>" />

    <button data-dough-clear-input-button type="reset" class="search__clear">
      <span class="icon icon--clear"></span>
      <span class="visually-hidden"><%= t('search_box.clear_hidden_text') %></span>
    </button>

    <button class="search__submit" type="submit">
      <span class="icon icon--search"></span>
      <span class="visually-hidden"><%= t('search_box.button_text') %></span>
    </button>
  </form>
<% end %>
```

## Possible Solution: #1 
```
#file: app/views/errors/show.html.erb

<% case @rescue_response 
 when :not_found %>
    <%= render partial: '_error_page_search', local: { error_type: 'not_found' } %>
<% when :internal_server_error %>
    <%= render partial: '_error_page_search', local: { error_type: 'internal_server_error' } %>
<% end %>
```

```
#file: app/views/errors/_error_page_search.html.erb

  <p class="error-page__search-text"><%= t(".#{error_type}.message") %></p>

  <form data-dough-component="ClearInput" action="/en/search" method="get" role="search" class="error-page__search search">
    <label class="visually-hidden" for="search"><%= t('search_box.label') %></label>
    <input data-dough-clear-input class="search__input" id="search" required type="search" name="query" autocomplete="off" placeholder="<%= t('.#{error_type}.search_placeholder') %>" />

    <button data-dough-clear-input-button type="reset" class="search__clear">
      <span class="icon icon--clear"></span>
      <span class="visually-hidden"><%= t('search_box.clear_hidden_text') %></span>
    </button>

    <button class="search__submit" type="submit">
      <span class="icon icon--search"></span>
      <span class="visually-hidden"><%= t('search_box.button_text') %></span>
    </button>
  </form>
```

## Possible Solution: #2
```
#file: app/views/errors/show.html.erb

 <p class="error-page__search-text"><%= t('.#{@rescue_response}.message') %></p>

  <form data-dough-component="ClearInput" action="/en/search" method="get" role="search" class="error-page__search search">
    <label class="visually-hidden" for="search"><%= t('search_box.label') %></label>
    <input data-dough-clear-input class="search__input" id="search" required type="search" name="query" autocomplete="off" placeholder="<%= t('.#{@rescue_response}.search_placeholder') %>" />

    <button data-dough-clear-input-button type="reset" class="search__clear">
      <span class="icon icon--clear"></span>
      <span class="visually-hidden"><%= t('search_box.clear_hidden_text') %></span>
    </button>

    <button class="search__submit" type="submit">
      <span class="icon icon--search"></span>
      <span class="visually-hidden"><%= t('search_box.button_text') %></span>
    </button>
  </form>
<% end %>
```